core-bundle/src/Resources/contao/models/PageModel.php

```diff
@@ -327,10 +336,19 @@ public function __set($strKey, $varValue)
 					case 'robots':
 						$htmlHeadBag->setMetaRobots($varValue);
 						break;
+				}
+			}
 
-					case 'noSearch':
-						$htmlHeadBag->setMetaRobots($varValue ? 'noindex,nofollow' : 'index,nofollow');
-						break;
+			if ('noSearch' === $strKey && $responseContext->has(JsonLdManager::class))
+			{
+				/** @var JsonLdManager $jsonLdManager */
+				$jsonLdManager = $responseContext->get(JsonLdManager::class);
+
+				if ($jsonLdManager->getGraphForSchema(JsonLdManager::SCHEMA_CONTAO)->has(ContaoPageSchema::class))
+				{
+					/** @var ContaoPageSchema $schema */
+					$schema = $jsonLdManager->getGraphForSchema(JsonLdManager::SCHEMA_CONTAO)->get(ContaoPageSchema::class);
+					$schema->setNoSearch($varValue);
```

Comment:
```suggestion
					$schema->setNoSearch((bool) $varValue);
```
should we cast to boolean here?